pkg/aws/eks/eks.go

```diff
@@ -0,0 +1,360 @@
+
+	cloudcostexporter "github.com/grafana/cloudcost-exporter"
+	ec2client "github.com/grafana/cloudcost-exporter/pkg/aws/services/ec2"
+	pricingClient "github.com/grafana/cloudcost-exporter/pkg/aws/services/pricing"
+	"github.com/grafana/cloudcost-exporter/pkg/provider"
+)
+
+const (
+	subsystem = "eks"
+)
+
+// These were generated by analysing Grafana Labs spend in GCP and finding the ratio of CPU to Memory spend by instance type.
+// It's an imperfect approximation, but it's better than nothing.
+var cpuToCostRation = map[string]float64{
+	"Compute optimized": 0.88,
+	"Memory optimized":  0.48,
+	"General purpose":   0.65,
+}
+var clusterTags = []string{"cluster", "eks:cluster-name", "aws:eks:cluster-name"}
+
+var (
+	ErrRegionNotFound       = errors.New("no region found")
+	ErrInstanceTypeNotFound = errors.New("no instance type found")
+	ErrClientNotFound       = errors.New("no client found")
+	ListSpotPricesErr       = errors.New("error listing spot prices")
+	GeneratePricingMapErr   = errors.New("error generating pricing map")
+)
+
+var (
+	InstanceCPUHourlyCostDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(cloudcostexporter.MetricPrefix, subsystem, "instance_cpu_usd_per_core_hour"),
+		"The cpu cost a compute instance in USD/(core*h)",
+		[]string{"instance", "region", "family", "machine_type", "cluster", "price_tier", "provider"},
+		nil,
+	)
+	InstanceMemoryHourlyCostDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(cloudcostexporter.MetricPrefix, subsystem, "instance_memory_usd_per_gib_hour"),
+		"The memory cost of a compute instance in USD/(GiB*h)",
+		[]string{"instance", "region", "family", "machine_type", "cluster", "price_tier", "provider"},
+		nil,
+	)
+)
+
+type Attributes struct {
+	Region            string `json:"regionCode"`
+	InstanceType      string `json:"instanceType"`
+	VCPU              string `json:"vcpu"`
+	Memory            string `json:"memory"`
+	InstanceFamily    string `json:"instanceFamily"`
+	PhysicalProcessor string `json:"physicalProcessor"`
+	Tenancy           string `json:"tenancy"`
+	MarketOption      string `json:"marketOption"`
+	OperatingSystem   string `json:"operatingSystem"`
+	ClockSpeed        string `json:"clockSpeed"`
+	UsageType         string `json:"usageType"`
+}
+
+type productTerm struct {
+	Product struct {
+		Attributes Attributes
+	}
+	Terms struct {
+		OnDemand map[string]struct {
+			PriceDimensions map[string]struct {
+				PricePerUnit map[string]string `json:"pricePerUnit"`
+			}
+		}
+	}
+}
+
+type Collector struct {
+	Region          string
+	Regions         []ec2Types.Region
+	Profile         string
+	Profiles        []string
+	ScrapeInterval  time.Duration
+	pricingMap      *StructuredPricingMap
+	pricingService  pricingClient.Pricing
+	ec2Client       ec2client.EC2
+	NextScrape      time.Time
+	ec2RegionClient map[string]ec2client.EC2
+}
+
+func (c *Collector) CollectMetrics(metrics chan<- prometheus.Metric) float64 {
+	//TODO implement me
+	panic("implement me")
```

Comment:
is here something todo?

Reply:
THe main thing is that the `CollectMetrics` interface is a leaky abstraction and should be deprecated 😅 For now I'll just return 1 or whatever the appropriate method is.